mysql-test/t/mysqldump_gtid.inc

```diff
@@ -0,0 +1,69 @@
+--source include/have_log_bin.inc
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+--source include/have_debug_sync.inc
+
+--echo #################################################
+--echo # Testing mysqldump with options: $MYSQLDUMP_OPTS
+--echo #################################################
+
+--let sleep_sec = 5
+
+--connect(con1, localhost, root)
+
+--connection default
+CREATE TABLE t1 (a INT) ENGINE = InnoDB;
+INSERT INTO t1 VALUES (0);
+
+# Capture current gtid
+--let $gtid_start = `SELECT @@GLOBAL.GTID_EXECUTED`
+
+# Start mysqldump
+--let $file = $MYSQLTEST_VARDIR/tmp/dump.sql
+--let $start_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_start.sql
+--let $finish_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_finish.sql
+--exec_in_background $MYSQL_DUMP $MYSQLDUMP_OPTS test t1 --start-sql-file=$start_sql_file --finish-sql-file=$finish_sql_file >$file
```

Comment:
Is it possible to somehow get process id here to be able to wait for its termination at the end?

Reply:
I have introduced 2 generic inc files for 
1. starting process in background + collect its pid and 
2. waiting for pid to finish
Please check comments in start_proc_in_background.inc why it cannot be achieved with --exec_in_background command.
I think in general escaping characters when passing them via Perl to mysqltest layer could be fixed/improved, but if we decide to invest time there my suggestion is to handle it as the separate ticket/PR as it will need bigger rework/testing (and impact other tests). In general escape characters should be removed one by one by every layer, so 
1. having \\$abc in *.test file should cause passing \$abc to mysqltest
2. having \$abc in *.test file should cause passing $abc to mysqltest
3. having $abc in *.test file should substitue abc variable by its value and pass the value to mysqltest